FWCore/Reflection/test/run_checkClassVersionUpdate.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash -e
+
+SCRAM_TEST_NAME=TestFWCoreReflectionCheckClassVersionUpdate
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+
+# Create a new CMSSW dev area and build modified DataFormats/TestObjects in it
+NEW_CMSSW_BASE=$(/bin/pwd -P)/$CMSSW_VERSION
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+pushd $CMSSW_VERSION/src
+
+# Copy FWCore/Reflection code to be able to edit it to make ROOT header parsing to fail
+for DIR in ${CMSSW_BASE} ${CMSSW_RELEASE_BASE} ${CMSSW_FULL_RELEASE_BASE} ; do
+    if [ -d ${DIR}/src/FWCore/Reflection ]; then
+        mkdir FWCore
+        cp -Lr ${DIR}/src/FWCore/Reflection FWCore/
+        break
+    fi
+done
+if [ ! -e FWCore/Reflection ]; then
+    echo "Failed to symlink FWCore/Reflection from local or release area"
+    exit 1;
+fi
+
+# Modify the IntObject class to trigger a new version
+#
+# Just setting USER_CXXFLAGS for scram is not sufficient,because
+# somehow ROOT (as used by edmCheckClassVersion) still picks up the
+# version 3 of the class
+sed -i '1s/^/#define FWCORE_REFLECTION_TEST_INTOBJECT_V4\n/' FWCore/Reflection/test/stubs/TestObjects.h
+
+
+#Set env and build in sub-shel
+(eval $(scram run -sh) ; scram build -v -j $(nproc))
+
+popd
+
+# Prepend NEW_CMSSW_BASE's lib/src paths in to LD_LIBRARY_PATH and ROOT_INCLUDE_PATH
+export LD_LIBRARY_PATH=${NEW_CMSSW_BASE}/lib/${SCRAM_ARCH}:${LD_LIBRARY_PATH}
+export ROOT_INCLUDE_PATH=${NEW_CMSSW_BASE}/src:${ROOT_INCLUDE_PATH}
```

Comment:
Up to here this test script mimics https://github.com/cms-sw/cmssw/blob/master/IOPool/Input/test/testNoParentDictionary.sh

I couldn't come up with a simpler setup that would check the behavior of the `edmCheckClassVersion` when a class is updated (I tried a few but they didn't work).